edit/cms_functions.php

```diff
@@ -0,0 +1,459 @@
+ */
+function updateAbout($about) {
+    if (!isset($about['name']) || !isset($about['title']) || !isset($about['desc'])) {
+        return FALSE;
+    }
+    $name = filter_var($about['name'], FILTER_SANITIZE_STRING);
+    $title = filter_var($about['title'], FILTER_SANITIZE_STRING);
+    $desc = filter_var($about['desc'], FILTER_SANITIZE_STRING);
+    $stmt = getPDO()->prepare('UPDATE `about` SET `name` = :name, `title` = :title, `desc` = :desc');
+    $stmt->bindParam(':name',$name);
+    $stmt->bindParam(':title',$title);
+    $stmt->bindParam(':desc',$desc);
+    return $stmt->execute();
+}
+
+/*
+ * Gets the project data in the database under the given id.
+ *
+ * @param int id The unique ID of the target project.
+ *
+ * @return $array Returns an associative array of the project's name, type, description, link and image OR false if there was no
+ * project of that ID in the database.
+ */
+function getProjectFromDB(int $id) {
+    $stmt = getPDO()->prepare('SELECT `title`,`type`,`desc`,`image`,`link` FROM `projects` WHERE `id` = :id LIMIT 1');
+    $stmt->bindParam(':id',$id);
+    $stmt->execute();
+    $project = $stmt->fetch();
+    return $project;
+}
+
+
+/*
+ * Gets the project data from the POST and returns it as an associative array.
+ *
+ * @param array postData The post data as an array.
+ *
+ * @return $array Returns an associative array of the project's name, type, description, link and image.
+ */
+function getProjectDataFromPOST($postData) {
+    $project = [];
+    $project['title'] = filter_var($postData['title'], FILTER_SANITIZE_STRING);
+    $project['type'] = filter_var($postData['type'], FILTER_SANITIZE_STRING);
+    $project['desc'] = filter_var($postData['desc'], FILTER_SANITIZE_STRING);
+    $project['img'] = filter_var($postData['img'], FILTER_SANITIZE_STRING);
+    $project['link'] = filter_var($postData['link'], FILTER_SANITIZE_STRING);
+    return $project;
+}
+
+
+/*
+ * Gets the contact info entry from the POST and returns it as an associative array.
+ *
+ * @param array postData The post data as an array.
+ *
+ * @return $array Returns an associative array of the project's name, type, description, link and image.
+ */
+function getContactInfoFromPOST($postData) {
+    $contact = [];
+    $contact['icon_id'] = filter_var($postData['icon_id'], FILTER_SANITIZE_STRING);
+    $contact['link'] = filter_var($postData['link'], FILTER_SANITIZE_STRING);
+    $contact['text'] = filter_var($postData['text'], FILTER_SANITIZE_STRING);
+    return $contact;
+}
+
+/*
+ * Adds a new project to the database.
+ *
+ * @param array $project An associative array containing all the project information.
+ *
+ * @return Returns TRUE if the project was added successfully, FALSE if otherwise.
+ */
+function addProjectToDatabase($project) {
+    $stmt = getPDO()->prepare('INSERT INTO `projects`(`title`, `type`, `desc`,`image`,`link`) VALUES(:title, :type, :desc, :image, :link)');
```

Comment:
This functions allows me to enter null values in all fields.  Is this what you intended?

Reply:
fixed in df936f7